### ISSUE TITLE
Add --ci flag for running in GitLab CI

### ISSUE THREAD

**Issue by hawkrives** [StoDevX/cs251-toolkit#110]
```
$ cs251tk -n --partial --record $(ls students/"$CI_PROJECT_NAME"/ | grep 'hw') | tee result.md

$ if grep -F '[' result.md; then cat logs/log-hw1.md; else echo "no issues"; fi
```

That should _roughly_ be equivalent to 

```
$ cs251tk --ci
```

That is, `--ci` should be equivalent to `--no-progress`, `--no-update`, `--partials`, `--no-check`(?); it should also auto-discover all submitted homework folders with a glob, and automatically record them.

Any failing homeworks should have their logs printed to STDOUT; any partial homeworks should also be printed to STDOUT, with a list of what is missing / present. (I think this is partially present in the normal record output, but we should highlight it some more.)

---

This way, a GitLab CI config that will automatically email the student after a push with what failed for them could looks like this:

```yaml
image: 'stodevx/cs251-toolkit:v2.4.3'
build:
  stage: build
  before_script:
    - shopt -s extglob
    - mkdir -p ./students/"$CI_PROJECT_NAME"
    - mv !(students) ./students/"$CI_PROJECT_NAME"
    - echo "$CI_PROJECT_NAME" > students.txt
    - git clone https://github.com/Jedmeyer/cs253-specs.git ./data
  script:
    - cs251tk --ci
```

**Comment by hawkrives** [comment]
As a side note, @JedMeyer and I came up with this config to run cs251tk in a GitLab CI runner, and email the results to the student:

```yaml
image: 'stodevx/cs251-toolkit:v2.4.3'
build:
  stage: build
  before_script:
    - shopt -s extglob
    - mkdir -p ./students/"$CI_PROJECT_NAME"
    - mv !(students) ./students/"$CI_PROJECT_NAME"
    - echo "$CI_PROJECT_NAME" > students.txt
    - git clone https://github.com/Jedmeyer/cs253-specs.git ./data
  script:
    - cs251tk -n --partial --record $(ls students/"$CI_PROJECT_NAME"/ | grep 'hw') | tee results.out
    - >
      if grep -F '[' results.out;
      then
        INCOMPLETES=$(perl -l -ne 'print $1 while /\[31m(.+?).\[/g' < results.out);
        for hw in $INCOMPLETES; do
          cat logs/log-hw$hw.md
          for i in $(seq 1 20); do
            echo
          done
        done
        echo "Incomplete homeworks:"
        for hw in $INCOMPLETES; do
          echo "homework $hw"
        done
        echo "See StoGit website for full details."
        exit 1
      else
        echo "no issues"
      fi
```

**Comment by maxnz** [comment]
Using `cs251tk -n --partial --record $(ls students/"$CI_PROJECT_NAME"/ | grep 'hw') --students "$CI_PROJECT_NAME" | tee results.out` also works, removing the need for the `echo "$CI_PROJECT_NAME" > students.txt` line

**Comment by hawkrives** [comment]
Closed by #113
